### PR TITLE
BagelDB (bageldb.ai), VectorStore integration.

### PR DESCRIPTION
**Pull request by BidhanRoy**
-   **Description**: [BagelDB](bageldb.ai) a collaborative vector database. Integrated the bageldb PyPi package with langchain with related tests and code.

  - **Issue**: Not applicable.
  - **Dependencies**: `betabageldb` PyPi package.
  - **Tag maintainer**: @rlancemartin, @eyurtsev, @baskaryan
  - **Twitter handle**: bageldb_ai (https://twitter.com/BagelDB_ai)
  
We ran `make format`, `make lint` and `make test` locally.

Followed the contribution guideline thoroughly https://github.com/hwchase17/langchain/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by Towhid1** [comment]
We tried it on our end and working properly.

File: libs/langchain/tests/integration_tests/vectorstores/test_bagel.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    )
+
+    updated_content = "updated bagel doc"
+    updated_doc = Document(page_content=updated_content, metadata={"page": "0"})
+    docsearch.update_document(document_id=document_id, document=updated_doc)
+    output = docsearch.similarity_search(updated_content, k=1)
+    assert output == [Document(page_content=updated_content, metadata={"page": "0"})]
+
+
+def main() -> None:
```
Comment (leo-gan) [inline libs/langchain/tests/integration_tests/vectorstores/test_bagel.py:172]: I'm not sure L:172-L:187 code should be here. Isn't it a part of development that was not removed?

**Comment by leo-gan** [comment]
Is it possible to add an Integration Card and a usage example?
It could be like this: https://python.langchain.com/docs/integrations/providers/annoy - docs/extras/integrations/providers/annoy.mdx 
and 
https://python.langchain.com/docs/integrations/vectorstores/annoy - docs/extras/integrations/vectorstores/annoy.ipynb

**Comment by baskaryan** [comment]
> Is it possible to add an Integration Card and a usage example?
> It could be like this: https://python.langchain.com/docs/integrations/providers/annoy - docs/extras/integrations/providers/annoy.mdx
> and
> https://python.langchain.com/docs/integrations/vectorstores/annoy - docs/extras/integrations/vectorstores/annoy.ipynb

yea some docs would be awesome!

File: libs/langchain/tests/integration_tests/vectorstores/test_bagel.py (hunk 1 above)
Comment (Towhid1) [reply libs/langchain/tests/integration_tests/vectorstores/test_bagel.py:172]: Yeah. Fixing it.

**Comment by BidhanRoy** [comment]
@baskaryan @leo-gan updated according to the review.

**Comment by baskaryan** [comment]
looks like you need to merge master in, but otherwise lgtm!

**Comment by Towhid1** [comment]
> looks like you need to merge master in, but otherwise lgtm!

Merged

**Review by leo-gan — APPROVED** [review]
lgtm  Thanks!

**Review by Towhid1 — APPROVED** [review]

**Comment by baskaryan** [comment]
believe you need to update poetry.lock file `poetry lock --no-update`

**Comment by baskaryan** [comment]
if you want you can merge #9081 into this branch and that should fix

**Comment by Towhid1** [comment]
> if you want you can merge #9081 into this branch and that should fix

Plz share how to merge it. I tried but failed to do it. @baskaryan

**Comment by Towhid1** [comment]
> if you want you can merge #9081 into this branch and that should fix

@baskaryan @leo-gan Plz take a look. I think it's ok to merge now.

**Comment by baskaryan** [comment]
looks great, thanks @Towhid1 @BidhanRoy!

**Pull request merged**